projects/compilation.vala

```diff
@@ -0,0 +1,209 @@
+                    workaround_038 (_ctx, _sources.values);
+                }
+                // generate a new code context 
+                _ctx = new Vala.CodeContext () { keep_going = true };
+                Vala.CodeContext.push (_ctx);
+                dirty = false;
+
+                // string version = Config.libvala_version;
+                _ctx.report = new Reporter ();
+
+                // compiler flags and defines
+                _ctx.profile = profile;
+                switch (profile) {
+                    case Vala.Profile.POSIX:
+                        _ctx.add_define ("POSIX");
+                        break;
+                    default: // case Vala.Profile.GOBJECT:
+                        _ctx.add_define ("GOBJECT");
+                        break;
+                }
+
+                _ctx.experimental = experimental;
+                _ctx.experimental_non_null = experimental_non_null;
+                _ctx.abi_stability = abi_stability;
+
+                _ctx.vapi_directories = _vapi_dirs.to_array ();
+                _ctx.gir_directories = _gir_dirs.to_array ();
+                _ctx.metadata_directories = _metadata_dirs.to_array ();
+                _ctx.gresources_directories = _gresources_dirs.to_array ();
+
+                // packages
+                _ctx.add_external_package ("glib-2.0");
+                _ctx.add_external_package ("gobject-2.0");
+                foreach (var package in _packages)
+                    _ctx.add_external_package (package);
+
+                foreach (TextDocument doc in _sources.values) {
+                    doc.file.context = _ctx;
+                    _ctx.add_source_file (doc.file);
+                    // clear all using directives (to avoid "T ambiguous with T" errors)
+                    doc.file.current_using_directives = new Vala.ArrayList<Vala.UsingDirective> ();
+                    // add default using directives for the profile
+                    if (profile == Vala.Profile.POSIX) {
+                        // import the Posix namespace by default (namespace of backend-specific standard library)
+                        var ns_ref = new Vala.UsingDirective (new Vala.UnresolvedSymbol (null, "Posix", null));
+                        doc.file.add_using_directive (ns_ref);
+                        _ctx.root.add_using_directive (ns_ref);
+                    } else if (profile == Vala.Profile.GOBJECT) {
+                        // import the GLib namespace by default (namespace of backend-specific standard library)
+                        var ns_ref = new Vala.UsingDirective (new Vala.UnresolvedSymbol (null, "GLib", null));
+                        doc.file.add_using_directive (ns_ref);
+                        _ctx.root.add_using_directive (ns_ref);
+                    }
+
+                    // clear all comments from file
+                    doc.file.get_comments ().clear ();
+                    assert (doc.file.get_comments ().size == 0);
```

Comment:
are these assertions necessary?

Reply:
They're from a very long time ago, and probably not.

Reply:
can you delete them

Reply:
Sure.